lib/spack/spack/test/bindist.py

```diff
@@ -116,8 +82,14 @@ def config_directory(tmpdir_factory):
 
 
 @pytest.fixture(scope='function')
-def default_config(tmpdir_factory, config_directory, monkeypatch):
-
+def default_config(
+        tmpdir_factory, config_directory, monkeypatch,
+        install_mockery_mutable_config
+):
+    # This fixture depends on install_mockery_mutable_config to ensure
+    # there is a clear order of initialization. The substitution of the
+    # config scopes here is done on top of the substitution that comes with
+    # install_mockery_mutable_config
```

Comment:
>  why it's useful or necessary to have another substitution of config scopes here

This I don't know :slightly_smiling_face:  

I just noticed that the `default_config` fixture was conflicting with another fixture being used, i.e. `install_mockery`. The conflict is somehow hidden, since one must also know that:
-  `install_mockery` depends on another fixture (called `config`) that swaps in a mock configuration used only in tests
-  since `install_mockery` and `default_config` don't depend on each other, the initialization order was undefined

Here I established a clear order of initialization by making `default_config` dependent on `install_mockery`. Maybe further simplifications may be done here, but I didn't want to change the semantics of the tests.

Reply:
> * since `install_mockery` and `default_config` don't depend on each other, the initialization order was undefined

Ok, this makes sense and clears up my misunderstanding of your PR description comment.

Thanks for the explanation!

Reply:
I needed a default and a non-default config for the directory layout. The only way I could get it to work was the hackery I did.